File: php/src/Advertisements/User/Domain/ValueObjects/Role.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisements\User\Domain\ValueObjects;

enum Role
{
    case SUPERVISOR;
    case ADMIN;
    case MEMBER;

    public static function fromString(string $role): self
    {
        return match ($role) {
            'supervisor' => self::SUPERVISOR,
            'admin' => self::ADMIN,
            'member' => self::MEMBER,
        };
    }

    public function value(): string
    {
        return match ($this) {
            self::SUPERVISOR => 'supervisor',
            self::ADMIN => 'admin',
            self::MEMBER => 'member',
        };
    }

    public function isAdmin(): bool
    {
        return $this === self::ADMIN;
    }
}

File: kotlin/src/main/kotlin/advertisements/user/domain/value_object/Role.kt
package advertisements.user.domain.value_object

enum class Role {
    SUPERVISOR,
    ADMIN,
    MEMBER;

    companion object {
        fun fromString(role: String): Role {
            return when (role.lowercase()) {
                "supervisor" -> SUPERVISOR
                "admin" -> ADMIN
                "member" -> MEMBER
                else -> throw IllegalArgumentException("Unknown role: $role")
            }
        }
    }

    fun value(): String {
        return when (this) {
            SUPERVISOR -> "supervisor"
            ADMIN -> "admin"
            MEMBER -> "member"
        }
    }

    fun isAdmin(): Boolean {
        return this == ADMIN
    }
}
